pkg/agent/monitortool/latency_store.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitortool
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ip"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+// LatencyStore is a store for latency information of connections between Nodes.
+type LatencyStore struct {
+	// Lock for the latency store
+	mutex sync.RWMutex
+
+	// Whether the agent is running in networkPolicyOnly mode
+	isNetworkPolicyOnly bool
+	// The map of Node IP to latency entry, it will be changed by latency monitor
+	nodeIPLatencyMap map[string]*NodeIPLatencyEntry
+	// The map of Node name to Node IP, it will be changed by Node watcher
+	// If the agent is running in networkPolicyOnly mode, the value will be the transport IP of the Node
+	// If the agent is running in normal mode, the value will be the gateway IP of the Node
+	nodeTargetIPsMap map[string][]net.IP
+}
+
+// NodeIPLatencyEntry is the entry of the latency map.
+type NodeIPLatencyEntry struct {
+	// The timestamp of the last send packet
+	LastSendTime time.Time
+	// The timestamp of the last receive packet
+	LastRecvTime time.Time
+	// The last valid rtt of the connection
+	LastMeasuredRTT time.Duration
+}
+
+// NewLatencyStore creates a new LatencyStore.
+func NewLatencyStore(isNetworkPolicyOnly bool) *LatencyStore {
+	store := &LatencyStore{
+		nodeIPLatencyMap:    make(map[string]*NodeIPLatencyEntry),
+		nodeTargetIPsMap:    make(map[string][]net.IP),
+		isNetworkPolicyOnly: isNetworkPolicyOnly,
+	}
+
+	return store
+}
+
+// getNodeIPLatencyEntry returns the NodeIPLatencyEntry for the given Node IP
+// Now it used for testing purpose and it is not exported, so we just return a pointer
+func (l *LatencyStore) getNodeIPLatencyEntry(key string) (*NodeIPLatencyEntry, bool) {
```

Comment:
can we rename the `key` parameter to `nodeIP` for all these methods?

Reply:
OK,  I have changed them.